servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/KeepAliveManager.java

```diff
@@ -117,10 +116,15 @@ protected void channelIdle(final ChannelHandlerContext ctx, final IdleStateEvent
         }
         this.channel = channel;
         this.scheduler = scheduler;
-        if (keepAlivePolicy != null) {
+        // Before 0.42.30, H2ProtocolConfig.keepAlivePolicy() was @Nullable. For backward compatibility, we keep
+        // tolerance for null values.
+        if (keepAlivePolicy != null) {  // FIXME: 0.43.x - consider removing null check
+            // KeepAlivePolicy with idlenessThresholdNanos <= 0 disables PINGs, but allows configuring
+            // pingAckTimeoutNanos for graceful closure (GO_AWAY).
             disallowKeepAliveWithoutActiveStreams = !keepAlivePolicy.withoutActiveStreams();
             pingAckTimeoutNanos = keepAlivePolicy.ackTimeout().toNanos();
-            pingWriteCompletionListener = future -> {
+            final long idlenessThresholdNanos = keepAlivePolicy.idleDuration().toNanos();
+            pingWriteCompletionListener = idlenessThresholdNanos > 0 ? future -> {
```

Comment:
Didn't move `pingWriteCompletionListener` under a single `if` block to avoid indentation change. Wanted to keep git changes minimal to preserve the history.